### ISSUE TITLE
 handleLoadPackage is called 3 times in one process

### ISSUE THREAD

**Issue by zeroleo12345** [rovo89/XposedBridge#225]
In what situation, the function handleLoadPackage may be called 3 times by 1 process?
the below code is executed 3 times, and the PID is the same. Feel confused.
Just some app occurred this problem.
```
public class Main implements IXposedHookLoadPackage {
    @Override
    public void handleLoadPackage(final LoadPackageParam loadPackageParam) throws Throwable {
        if ( ! loadPackageParam.packageName.equals( TARGET_PACKAGE_NAME ) ) return;
        if ( ! loadPackageParam.processName.equals( TARGET_PROCESS_NAME ) ) return;
        XLog.i("appInfo.processName:%s, processName:%s, PID:%d", loadPackageParam.appInfo.processName, loadPackageParam.processName, android.os.Process.myPid());
    }
```

**Comment by Massi-X** [comment]
Can you please be more clear? When this happens? With what applications? Maybe it's because the app is really called 3 times? Logs?

**Comment by rovo89** [comment]
Was the process name the same? This function might indeed be called multiple times in the same process if multiple packages/APKs are loaded in the same process. That can be the case for some manifest settings or when explicitely requesting the code of another app.

See also the `isFirstApplication` attribute: http://api.xposed.info/reference/de/robv/android/xposed/callbacks/XC_LoadPackage.LoadPackageParam.html

**Comment by zeroleo12345** [comment]
@rovo89 yes, **process name, packageName, PID** are both same in these 3 calling. 
Do you mean that use DexClassLoader() or PathClassLoader() to load other apk file,  will trigger *handleLoadPackage()* being called?
I want to learn more, and want do some test for reproducing. Please.

I also print the `isFirstApplication` attribute, but it both true in these 3calling.

**Comment by wowwow21098** [comment]
IsFirstApplication
